src/cmake-tools.ts

```diff
@@ -481,18 +484,31 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       vscode.window.showErrorMessage(localize('no.source.directory.found', 'You do not have a source directory open'));
       break;
     case CMakePreconditionProblems.MissingCMakeListsFile:
+      telemetryEvent = "partialActivation";
+      telemetryProperties["ignoreCMakeListsMissing"] = this.workspaceContext.state.ignoreCMakeListsMissing.toString();
+
       if (!this.workspaceContext.state.ignoreCMakeListsMissing) {
         const quickStart = localize('quickstart.cmake.project', 'Create');
         const changeSetting = localize('edit.setting', 'Locate');
         const ignoreCMakeListsMissing = localize('ignore.activation', "Don't show again");
-        const result = await vscode.window.showErrorMessage(
+        const showToast: boolean | undefined = await expShowToast();
+
+        telemetryProperties["missingCMakeListsPopupType"] = showToast ? "toastCreateLocateIgnore" : "selectFromAllCMakeLists";
+
+        const result = showToast ? await vscode.window.showErrorMessage(
             localize('missing.cmakelists', 'CMakeLists.txt was not found in the root of the folder \'{0}\'. How would you like to proceed?', this.folderName),
-            quickStart, changeSetting, ignoreCMakeListsMissing);
+            quickStart, changeSetting, ignoreCMakeListsMissing) : changeSetting;
+
+        if (showToast) {
+          telemetryProperties["missingCMakeListsUserAction"] = result || "dismissed";
```

Comment:
I like this, but we'll end up with localized properties which will be hard to aggregate. Let's get non-localized strings here.